### PR TITLE
sync with new OCP\IAppConfig

### PR DESCRIPTION
**Pull request by ArtificialOwl**
Requested enhancement: https://github.com/nextcloud/server/pull/41755#issuecomment-1909578250

### CODE REVIEW

File: lib/private/AppFramework/DependencyInjection/DIContainer.php [hunk 1]
```diff
@@ -345,7 +345,8 @@ public function __construct(string $appName, array $urlParams = [], ServerContai
 		$this->registerService(IAppConfig::class, function (ContainerInterface $c) {
 			return new OC\AppFramework\Services\AppConfig(
 				$c->get(IConfig::class),
```
Comment (nickvergessen) [inline lib/private/AppFramework/DependencyInjection/DIContainer.php:347]: this is still needed?

Comment (nickvergessen) [reply lib/private/AppFramework/DependencyInjection/DIContainer.php:347]: Ah yes for user preferences

**Review by nickvergessen — COMMENTED** [review]
It's the right time to add unit tests :-X

File: lib/private/AppFramework/DependencyInjection/DIContainer.php (hunk 1 above)
Comment (ArtificialOwl) [reply lib/private/AppFramework/DependencyInjection/DIContainer.php:347]: I think this is needed because you register the current app name ?

**Review by juliusknorr — APPROVED** [review]

File: lib/private/AppFramework/Services/AppConfig.php [hunk 2]
```diff
@@ -26,39 +26,321 @@
...
+	): bool {
+		return $this->appConfig->setValueArray($this->appName, $key, $value, $lazy, $sensitive);
+	}
+
+	/**
+	 * @param string $key
+	 * @param string $default
+	 *
+	 * @since 20.0.0
+	 * @deprecated use {@see getAppValueString()}
```
Comment (nickvergessen) [inline lib/private/AppFramework/Services/AppConfig.php:234]:
```suggestion
	 * @deprecated 29.0.0 use {@see getAppValueString()}
```

File: lib/private/AppFramework/Services/AppConfig.php [hunk 3]
```diff
@@ -26,39 +26,321 @@
...
+		return $this->appConfig->getAllValues($this->appName, $key, $filtered);
 	}
 
+	/**
+	 * @inheritDoc
+	 *
+	 * @param string $key the key of the value, under which will be saved
+	 * @param string $value the value that should be stored
+	 * @since 20.0.0
+	 * @deprecated use {@see setAppValueString()}
```
Comment (nickvergessen) [inline lib/private/AppFramework/Services/AppConfig.php:113]:
```suggestion
	 * @deprecated 29.0.0 use {@see setAppValueString()}
```

File: lib/public/AppFramework/Services/IAppConfig.php [hunk 4]
```diff
@@ -46,19 +100,196 @@ public function getAppKeys(): array ;
 	 * @param string $value the value that should be stored
 	 * @return void
 	 * @since 20.0.0
+	 * @deprecated use {@see setAppValueString()}
```
Comment (nickvergessen) [inline lib/public/AppFramework/Services/IAppConfig.php:103]:
```suggestion
	 * @deprecated 29.0.0 use {@see setAppValueString()}
```

File: lib/public/AppFramework/Services/IAppConfig.php [hunk 5]
```diff
@@ -46,19 +100,196 @@ public function getAppKeys(): array ;
...
+
 	/**
 	 * Looks up an app wide defined value
 	 *
 	 * @param string $key the key of the value, under which it was saved
 	 * @param string $default the default value to be returned if the value isn't set
+	 *
 	 * @return string the saved value
 	 * @since 20.0.0
+	 * @deprecated use {@see getAppValueString()}
```
Comment (nickvergessen) [inline lib/public/AppFramework/Services/IAppConfig.php:214]:
```suggestion
	 * @deprecated 29.0.0 use {@see getAppValueString()}
```

File: lib/public/AppFramework/Services/IAppConfig.php (hunk 4 above)
Comment (nickvergessen) [reply lib/public/AppFramework/Services/IAppConfig.php:103]: Should add the version so we know when we can throw them out =)

File: lib/private/AppFramework/Services/AppConfig.php [hunk 6]
```diff
@@ -26,39 +26,321 @@
...
+	 * @inheritDoc
+	 *
+	 * @param string $key config key
+	 * @param bool|null $lazy TRUE to search within lazy loaded config, NULL to search within all config
+	 *
+	 * @return bool TRUE if key exists
+	 * @since 29.0.0
+	 */
+	public function hasAppKey(string $key, ?bool $lazy = false): bool {
+		return $this->appConfig->hasKey($this->appName, $key);
```
Comment (nickvergessen) [inline lib/private/AppFramework/Services/AppConfig.php:65]:
```suggestion
		return $this->appConfig->hasKey($this->appName, $key, $lazy);
```

File: lib/private/AppFramework/Services/AppConfig.php [hunk 7]
```diff
@@ -26,39 +26,321 @@
...
+	/**
+	 * @param string $key config key
+	 * @param bool|null $lazy TRUE to search within lazy loaded config, NULL to search within all config
+	 *
+	 * @return bool
+	 * @throws AppConfigUnknownKeyException if config key is not known
+	 * @since 29.0.0
+	 */
+	public function isSensitive(string $key, ?bool $lazy = false): bool {
+		return $this->appConfig->isSensitive($this->appName, $key);
```
Comment (nickvergessen) [inline lib/private/AppFramework/Services/AppConfig.php:77]:
```suggestion
		return $this->appConfig->isSensitive($this->appName, $key, $lazy);
```

File: lib/private/AppFramework/Services/AppConfig.php [hunk 8]
```diff
@@ -26,39 +26,321 @@
...
+	 * @param string $key
+	 * @param string $default
+	 *
+	 * @since 20.0.0
+	 * @deprecated 29.0.0 use {@see getAppValueString()}
+	 * @return string
+	 */
 	public function getAppValue(string $key, string $default = ''): string {
-		return $this->config->getAppValue($this->appName, $key, $default);
+		/** @psalm-suppress InternalMethod */
```
Comment (nickvergessen) [inline lib/private/AppFramework/Services/AppConfig.php:238]:
```suggestion
		/** @psalm-suppress InternalMethod */
		/** @psalm-suppress UndefinedInterfaceMethod */
```

**Review by nickvergessen — COMMENTED** [review]
(Sorry for the confusion, as per chat the tests cover "core Appconfig" and should therefore be moved there, the Service is not really usefully testable)

**Review by nickvergessen — APPROVED** [review]

**Comment by ChristophWurst** [comment]
new APIs mandate developer docs

**Pull request merged**